modules/gapi/src/streaming/onevpl/accelerators/dx11_alloc_resource.hpp

```diff
@@ -0,0 +1,151 @@
+#ifndef GAPI_STREAMING_ONEVPL_ACCEL_DX11_ALLOC_RESOURCE_HPP
+#define GAPI_STREAMING_ONEVPL_ACCEL_DX11_ALLOC_RESOURCE_HPP
+
+#include <map>
+
+#include "opencv2/gapi/own/exports.hpp" // GAPI_EXPORTS
+#include <opencv2/gapi/util/compiler_hints.hpp>
+
+#ifdef HAVE_ONEVPL
+#include <vpl/mfxvideo.h>
+#include "streaming/onevpl/accelerators/utils/elastic_barrier.hpp"
+#include "streaming/onevpl/utils.hpp"
+
+#ifdef HAVE_DIRECTX
+#ifdef HAVE_D3D11
+#pragma comment(lib,"d3d11.lib")
+
+#define D3D11_NO_HELPERS
+#define NOMINMAX
+#include <d3d11.h>
+#include <d3d11_4.h>
+#include <codecvt>
+#include "opencv2/core/directx.hpp"
+#ifdef HAVE_OPENCL
+#include <CL/cl_d3d11.h>
+#endif // HAVE_OPENCL
+#undef D3D11_NO_HELPERS
+#undef NOMINMAX
+
+namespace cv {
+namespace gapi {
+namespace wip {
+namespace onevpl {
+
+class SharedLock;
+struct GAPI_EXPORTS LockAdapter {
+    LockAdapter(mfxFrameAllocator origin_allocator);
+
+    // GAPI_EXPORTS for tests
```

Comment:
Is it the right place for this comment?
I'd prefer it above `struct GAPI_EXPORTS LockAdapter`